### PR TITLE
[WIP] Bot interaction

### PR DESCRIPTION
**Pull request by filipevarjao**
[add] Sphinx can respond to commands like save, search, help

### CODE REVIEW

File: lib/sphinx_rtm/messages.ex [hunk 1]
```diff
@@ -18,19 +18,31 @@ defmodule SphinxRtm.Messages do
...
+
+          {:search, content} ->
+            content = if content == "", do: text, else: content
 
-          reply ->
-            {:reply,
-             "You asked for \"#{text}\" but I have no answer, but I found it: \n #{reply}"}
+            case SlackUtils.search(content, message.channel) do
+              nil ->
+                {:reply, "You asked for \"#{content}\" but I have no answer!"}
```
Comment (maiphuong-van) [inline lib/sphinx_rtm/messages.ex]: Maybe we can add some help here for user?

{:reply, "You asked for \"#{content}\" but I have no answer! Invoke `@sphinx [SAVE] [TEXT]` to save the question for future use!"}`

File: lib/sphinx_rtm/messages.ex [hunk 2]
```diff
@@ -18,19 +18,31 @@ defmodule SphinxRtm.Messages do
...
+          {:help, _} ->
+            {:reply, help()}
+
+          {:save, content} ->
+            reply =
+              message
+              |> Map.put(:text, content)
+              |> save_question()
+
+            {:reply, reply}
```
Comment (maiphuong-van) [inline lib/sphinx_rtm/messages.ex]: :no_reply or {:reply, "You're question is saved"}

File: lib/sphinx_rtm/messages.ex [hunk 3]
```diff
@@ -18,19 +18,31 @@ defmodule SphinxRtm.Messages do
...
-          reply ->
-            {:reply,
-             "You asked for \"#{text}\" but I have no answer, but I found it: \n #{reply}"}
+            case SlackUtils.search(content, message.channel) do
+              nil ->
+                {:reply, "You asked for \"#{content}\" but I have no answer!"}
+
+              reply ->
+                {:reply,
+                 "You asked for \"#{content}\" but I have no answer, but I found it: \n #{reply}"}
```
Comment (maiphuong-van) [inline lib/sphinx_rtm/messages.ex]: "You asked for \"#{content}\", you might find these helpful: \n #{reply}"

Comment (maiphuong-van) [reply lib/sphinx_rtm/messages.ex]: Just some text suggestion 😄

Comment (filipevarjao) [reply lib/sphinx_rtm/messages.ex]: great

**Review by maiphuong-van — APPROVED** [review]

**Pull request merged**